### DISCUSSION TITLE
Problems with schedules

### DISCUSSION THREAD

**Discussion by lockeandev** [tagyoureit/nodejs-poolController#502] [Q&A]
I've been having problems with schedules firing consistently ever since I updated my intellicenter i8ps to the latest firmware (1.064). 

I only have two circuits I regularly turn on "pump" and "pool". I had two schedules that would turn on at 8:00 am and run until 8:00pm. sometime the pump circuit would turn on, sometimes the pool circuit. I though maybe there was some latency with the OCP sending the commands to turn the features on, so I offset the schedules by 5 mins. that didnt seem to help at all. Next I've tried creating a circuit group with the "pump" and "pool" circuits included and then created a 1 schedule to turn that circuit on. same result, still hit or miss to get them to both turn on. 

Anyone else see this behavior?

**Comment by rstrouse** [comment]
What would a pump circuit do?  I can't see any purpose for a separate circuit being set up for a pump.  njsPC is passive in this instance it simply waits around for IntelliCenter to tell it which circuits and/or schedules have been activated.

**Reply by tagyoureit** [reply]
You would probably find that you could completely turn off njsPC and the i8ps would show the same behavior.  Probably not an njsPC issue...

**Reply by lockeandev** [reply]
I have an intelliflo vsf and I have various pump features circuits set up for various rpm/gpm rates. So when I want to run the pool and have the acid pump (controlled by rem) and ultra temp heat pump, and intellichlor active I have to turn on the "pool" circuit and the "pump 30gpm" circuit.  At least that the best way I've found to run it. 

And I agree that the intellicenter would probably handle the schedules the same way without the njsPC . The new firmware stinks. I regret purchasing the intellicenter and wish I would have known about nixie before I did. I was mostly wondering if anyone with intellicenter OCP were experiencing the same behavior and if they found a better way to get schedules working.

**Comment by rstrouse** [comment]
Oh I see you have set up a circuit for the pump speed.  I too have IntelliCenter and rarely use the UI for anything.  Do you have the pool circuit assigned to your filter pump?  The IntelliChlor requires the filter pump to be running and the pool circuit to be engaged.  Typically you would set it up so that the pool speed is the minimum speed required to run the chemistry equipment.  Then set up a feature with increased speed to run your pool cleaner.  All of my water features are on their own pumps but solar and heater increases the speed according to their needs.  In the end I don’t have any circuits specific to any pump speed.

**Reply by lockeandev** [reply]
I guess a little more about my system would be helpful. I have an intellicenter i8ps, but I only have one body of water - "pool". I have one pump that runs 3 features. When just the filter pump is running, the intellivalve for the return is open and my 3 valves for my 3 sets of features are shut. Then each of the features is set as a spillway which causes the system to close the return valve when the feature valve is opened. Each of the features is paired with a corresponding pump speed circuit.

I'm not sure I know how to assign the filter pump to the pool circuit - maybe I have already? I can turn my pool circuit on with the pump off but it doesn't do anything and the intellichlor and acid pump don't run because their flow switches are not triggered.  How do you get other circuits like the water features to increase the speed without using pump speed circuits like I'm doing. The only thing I've found is to create a circuit group with the valve for the feature and group it with a pump speed circuit.

Also I'm not sure how to turn the heater or intellichlor on other than just turning the pool circuit on.

**Reply by tagyoureit** [reply]
The main concept is that the "pool" circuit being on is that it signals to intellicenter that it enables the heater, chlorinator, etc.  You can still turn on the filter pump with any other circuit, but like you said, IC won't control the other equipment.

You assign the pump speeds to the circuit through the Pump Circuits.  You can do this in IC, too, but this is the njsPC view.
![2022-06-24_08-00-56](https://user-images.githubusercontent.com/1646840/175563424-1349df45-11b7-4791-93f1-378780927efc.png)

(Ignore my 'pool low').
1. Is my "pool" circuit.  It is the lowest speed and turns on control of the heater/chlorinator/REM acid pumps.
2. Is my "solar" circuit.  When I turn on the solar heater, this pump is increases to 1800rpm.
Likewise when I turn on my cleaner circuit, or jets, the pump speed will increase.

If you take a look at the drop down for circuits that you select, you'll see there is also specific virtual circuits for spillway, heater, freeze, and any other circuits/features/groups you create.

**Reply by rstrouse** [reply]
I can see where the spa functions would get in the way of operating normal functions for a single body.  Do you have the intake and return valves connected to anything?   It sounds like you do since you use the spillway function.  For the most part the intake valve will stubbornly get in the way of operating just your pool.  

There are also several functions that do not make sense and you should make sure you have them disabled.  The two that come to mind right away are valve delays and manual heat functions.

Have you considered selling the i8PS daughter card and attempting to get an i8P daughter card?  If not maybe we can figure out how to work around the valve issues but trying to make use of the intake and return valves would certainly require some duct tape.

**Reply by lockeandev** [reply]
@tagyoureit 
Thanks, I will have to look at my circuit setups again. I can't right now because my njPC has been going offline regularly. the whole rasppi is unresponsive. That's another issue but it seems to stay up and running for a doy or too and then the dashPanel becomes very slow to respond and eventually the whole thing goes offline.....

@rstrouse 
I don't think I can even find an i8p daughter card. I found one for $1000 on ebay but that's the best I can find and I'm not sure its worth it? it's too bad I can't just disable the spa side in software. 

I have the return valve connected to my returns so I can shut the returns off when I run the features. that's what enables my to use one pump to power the features. 

I've added some pictures of my plumbing. this was at about 80% complete, I didn't have the salt cell, heater or acid pump installed when these were taken. 

![image](https://user-images.githubusercontent.com/11232490/175575326-bb9616fa-cfe1-4514-b8b9-6c855394bffb.png)

![image](https://user-images.githubusercontent.com/11232490/175575377-a1887324-4a63-44fe-a8ab-46abaac7edb0.png)

**Reply by tagyoureit** [reply]
Interesting... what is an "endless pool current generator hydraulic motor"?  

It seems like you could run into trouble if your return and all three feature valves are turned off at the same time.  There should always be some path for the water to freely escape from the pump.  How are you ensuring that when your "spillway" is active at least one of the features is enabled?

**Reply by lockeandev** [reply]
The endless pool motor pumps hydraulic fluid to the propeller in a current generator. this is the product: https://www.endlesspools.com/products/fastlane-series/fastlane-pro

All of the water features are defined as spillway features which causes the ocp to close the return valve. so I think the only way it would deadhead is if someone manually closing a valve. I guess if a valve failed it could as well. The intelliflow has a pressure sensor and thermal protection that provides some protection in the event of deadhead. I know its not ideal. not really sure how else to do it without a separate pump for the water features.
also I wonder if switching to a i8p board would remove the spillway functionality?  typically its used for a spillway from a spa to a pool....

**Comment by rstrouse** [comment]
Create a backup of you setup and I will mirror the setup to see if there is anything odd about the settings.  Your pi should not be going offline.  One question though, are you hardwired or wifi to the pad?

Maybe the answer to the valves are to use either REM relays for your additional valves or use a valve expansion.  The problem with intake/return is that it is mode driven meaning it is trying to manage diversion between two bodies of water and the spa body has its own heater and functions that are goofy to main body.  If one could ignore the spa circuit then maybe you could forget it even exists.

**Reply by lockeandev** [reply]
I will try to reboot it tonight and get a backup.

I have a valve expansion, so I have plenty of room for actuators. not sure how to set it up without using the return valve. the only way I can ensure the states of the return and features are inverted it to have the ocp think the features are spillways and automatically close the return. right? is there a better way to do it?

**Reply by tagyoureit** [reply]
I think it's just fair to say you have a non-standard install.  With a typical shared body there are no possible paths for the water to take that could result in a dead-end, even if a valve fails.  But with your setup a dead-end is possible with a [feature] valve failure or return valve failure.  

The only "better" way to do it would have the features on their own pump, but it looks like that ship has sailed.

**Reply by lockeandev** [reply]
@rstrouse I tried to get in and get a backup for you to use to mirror. I can even get in via ssh. My router says the raspi gets online and within 1-3 minutes of being online. It it offline again.  Not sure what's wrong with it. Is there something I can pull from the SD card that might help diagnose it?

**Reply by rstrouse** [reply]
If you can get to the SD card.  Simply copy the nodejs-poolController/data directory.  I am looking for the poolConfig.json and poolState.json.  A copy of the nodejs-poolController/config.json would be helpful as well.   If you are not booting, if you have a monitor to hook up to the pi that might reveal the issue.

Another thing you can try is to image another SD card with nothing but the OS installed so you can see if there is something wrong with the pi.  It is very odd that even ssh is dropping out.  It would be one thing if njsPC was restarting or something like that but a hard software reset is quite another thing.

**Reply by tagyoureit** [reply]
Are you using a wireless connection?  You can get the signal level with `cat /proc/net/wireless`.  The level values will be 0 to -100.  The closer to 0 the better, but realistically a good connection will be in the -50ish range.  -60 to -70 is decent/poor.  Anything less than that will definitely give you trouble.

**Reply by lockeandev** [reply]
@tagyoureit I am using a wireless connection but I have unifi aps and a pretty strong signal at the pool equipment pad with my phone. I can't ssh into the raspi as it stands so I can't check signal strength. I think the pi might just be over heating. is there a log on the pi I should check somewhere? 

I was able to take the sd card out of the pi at the equipment pad, and boot it up into a spare pi, with no problem. I don't have the whitebox tentacle board or the sequent relay board that I have at the pool pad on the spare pi. not sure if that would matter. 

@rstrouse attached are the files for you to have a look. 
[njsPC config files.zip](https://github.com/tagyoureit/nodejs-poolController/files/9007323/njsPC.config.files.zip)

**Reply by tagyoureit** [reply]
> I think the pi might just be over heating.

See https://github.com/tagyoureit/nodejs-poolController/discussions/498.  A fan can work wonders and I mention tracking the temp through MQTT.

I took a look at your files.  Just a few observations:
1. your pH is high
2. I was expecting north jets and south jets to be assigned as a spillway, but I see they are assigned as magicstream laminars.  
3. You should uncheck the 'show as features' on the 'spillway switch' to make sure it is never activated outside of a circuit group
4. You should have the Pool circuit assigned to the pump with the lowest speed you want to run your pool.  In your current config, the OCP will think the pool should be on but never activate the pump unless you manually select a speed.  What happens every morning at 8:05am if a pump speed isn't selected?
5. Is your endless pool not controlled via the Intellicenter?

I don't see anything that would cause an issue with the Pi going offline, but if there's a needle in a haystack @rstrouse will find it.

**Reply by lockeandev** [reply]
Thanks for taking a look. 

1. My pH is actually 7.4. there is a problem with my probe see my additions to #493 
2. I think the jets have to be assigned as magicstream or I can't control the lights in them right? I have "show as feature" turned off I but have the north and south jets circuits grouped with the virtual spillway circuit and then have those paired groups exposed as features. So you can't turn the jets on without causing also turning the virtual spillway switch to turn on thereby closing the return valve.
3. Done.
4. This was the original purpose of this thread. My pool started turning on but no pump speed was being set because I think only one of the two circuits was turning on (i.e. "Pool" turned on, but "Pump 30" did not). So I think doing as you said will fix the original problem and ensuring the pump turns on every time the pool turns on.
5. I would love to control the endless pool via njsPC or something like MQTT. It uses some xbee wifi module to connect to some unknown cloud service where commands are relayed via a phone app. I've tried using Wireshark to try and capture the commands it is receiving via the web, but I'm a novice and can't make sense of anything I see in wireshark.

I went ahead and bought the sequent fan hat. Are there's special instructions to get it working with njsPC or REM?

**Reply by tagyoureit** [reply]
2.  Absolutely.  Just from the word jets I wasn't expecting it to be a light.  I would have labelled it laminars, personally.  
5.  What's the make/model?  Maybe we can find something in the manual.

> instructions for sequent smart fan

Of course.  ;). https://github.com/rstrouse/relayEquipmentManager/wiki/Sequent-Microsystems-Boards#sequent-smart-fan. The exponential curve algorithm is typically the best bet.

**Reply by lockeandev** [reply]
I used "jets" because my wife and kids don't appreciate the beauty or fully laminar flow like I do. They are just deck jets with lights to them. But I guess since the laminar circuits them selves are never exposed to the end user, just the laminars+spillway switch as "jets", I could have labeled the circuit as laminars. 

Are you asking for the make/model of the endless pool?  I don't think it is likely, but I have attached the manual for the unit. I tried using a broadlink RF universal rf remote to learn the commands from the rf remote included with the endless pool, but that was unsuccessful also. Pg 37 of the manual has the control board wiring diagram. and the wifi installation guide has actual pictures of the control board and shows the installation of the wifi add on. 

[FL Owners Manual.pdf](https://github.com/tagyoureit/nodejs-poolController/files/9012468/FL.Owners.Manual.pdf)

[WiFi-Fit-At-Home-App-Installation-Guide.pdf](https://github.com/tagyoureit/nodejs-poolController/files/9012469/WiFi-Fit-At-Home-App-Installation-Guide.pdf)


And thanks for the smart fan instructions. right now my HAT stack includes a sequent [4-relay board](https://sequentmicrosystems.com/products/raspberry-pi-relays-heavy-duty-hat), and a [whitebox T3](https://atlas-scientific.com/electrical-isolation/whitebox-t3/).   would you place the fan on the bottom of the stack, closer to the raspi?

**Reply by tagyoureit** [reply]
I'm not seeing much on the endless pool manuals for options to remote control.  Someone with skills to reverse engineer the network protocol could help, but that's beyond my skillset.  

As for the fan hat, you want it right next to the Pi so it can directly cool the cpu.

**Reply by rstrouse** [reply]
Hey what are you using to power the pi?

**Reply by lockeandev** [reply]
I'm using the [official rasp pi psu](https://www.adafruit.com/product/4298)

**Reply by rstrouse** [reply]
Is that psu perhaps failing?  One of the things that will cause repeated boot failures on the pi is low voltage.  It sounds like you were able to boot another pi with the same SD card just fine and your configuration files look fine as well.  I don't really suspect the T3 or the relay board.  On the off chance it is one of them before you restack your pi with a fan hat try it without them.

**Reply by lockeandev** [reply]
This rebooting and going unavailiable issue seems to resolve it self for a while but it is back again. We have almost two months of lower than normal temps with monsoonal weather, but now we are in another heatwave so my njs pc is going offline periodically again.  I finally restacked my pi with the fan hat last night. but then it went offline and back on again in the middle of the night. I don't think we had any power fluctuations or outages. so I'm back to wondering if it is the temps again or not. 

Is there a way to publish the cpu temp to influxdb or similar so I can see if there is a correlation between cpu temp and when it goes off line?

**Reply by tagyoureit** [reply]
We have an MQTT broker you can add to REM (home page, bottom) if you are using that.  MQTTExplorer has a history graph you can enable.  But there isn't a route from REM to Influx.

Alternatively, you could log to a file using something like this (plus date/time) to a file and view that file the next time you boot up. 
```
cpu=$(</sys/class/thermal/thermal_zone0/temp)
echo "$((cpu/1000)) c"
```

**Reply by lockeandev** [reply]
What am I doing wrong with the fan configuration. The CPU temp is showing 40C but the fan temp is only 4C?
![Screenshot_20220903_072632_Chrome](https://user-images.githubusercontent.com/11232490/188272477-934b4918-3e40-4c94-82f2-d23b840a055a.jpg)

**Reply by tagyoureit** [reply]
Hmm, can you install the [command line tools](https://github.com/SequentMicrosystems/SmartFan-rpi)?  Then run `fan 0 trd` and see what the temperature is reported directly from the board.  Likely it will be the same.

**Reply by lockeandev** [reply]
I did that and I get the message "Not Available for this hardware version". I noticed that my fan was not detected at address 3 as described in the wiki. It is on address 45. Not sure if that means anything.

**Reply by tagyoureit** [reply]
This is the fan you have, yes: https://sequentmicrosystems.com/products/smart-fan-for-raspberry-pi?

Why do you think it's on address 45?  You might need to manually add it to address 3.  Unless you changed the jumper on back, that's the default i2c address.  I've never see the command line tools display that message before.

**Reply by lockeandev** [reply]
Maybe it's a units thing? When I change to F it seems to work. When it's set to C, the fan temp reported in remchem is the same as whatever the fan speed is set to.
![Screenshot_20220904_103540_Chrome](https://user-images.githubusercontent.com/11232490/188324009-44aae2a9-d6dc-4099-a202-e761f52271f4.jpg)

**Reply by tagyoureit** [reply]
Can you see what revision your board is (and send a picture it possible)?  It's still very strange it is on another address and not responding to the command line tools.  I have the same smart fan on address 3 and it reports both C and F temperatures.

**Reply by lockeandev** [reply]
I will have to try to take a picture later but running the command line tool `fan -list: board` returns hardware version 4.0

**Reply by tagyoureit** [reply]
Interesting... I'll need to write up some things for you to test.  My board is `Hardware 01.01, Firmware 01.04`.  Let's start a new discussion (or issue) since this one was originally about schedules.  I'm guessing they changed the i2c address because anything <8 is now reserved for the system whereas it was <3 in the past.

**Comment by lockeandev** [comment]
So I have been running a few days without a separate "pump 30" circuit and instead have the "pool" circuit set with the minimum flow for the filter, chlorinator, heater and acid pump, ass suggested by @tagyoureit above. How over over the last 3 to 4 days, my initial problem persists, some times the schedule runs, some times it doesn't. I know it's not an njsPC issue, but @rstrouse, have you not seen similar issues with your intellicenter?

**Reply by tagyoureit** [reply]
My suggestions usually are ass... but have you tried a reset/power cycle on your IC to see if anything changes?

**Reply by rstrouse** [reply]
I have never witnessed IntelliCenter failing to start a schedule.  However, I do have some suggestions to try.  We will focus on how you have assigned circuits to the pump.  I also seem to remember that there is no spa here despite the i8PS daughter card.  Do you have a separate booster pump for the waterslide or is this all running off the main pump?

First assign your Pool Circuit to your filter pump.  Set it to the slowest speed when the pool is supposed to be engaged.  At this point it appears to be 12 hours starting at 8:05.  Using a circuit group for this is a bit dubious since engaging the pool circuit should start the pump at its lowest speed.  This appears to be 30gpm and not 20gpm.  

The reasons for doing this is because so much of the equipment is dependent upon the pool circuit engagement.  Then any time your pool circuit is on it will engage the pump to run at 30gpm.  If you turn off the pool circuit and run any of your other circuits or features it will run at their requisite speed.

I assume the reason for your use of the spillway circuit function is because you have 3 plumbing runs for water features on a single pump.  In this instance you are looking to route water from the normal pool returns to the slide and north/south jets.  The unfortunate part is that spillway has some special programming in IntelliCenter and is subject to the valve rotation delay functions.  I have a guess as to the incarnations it goes through but ultimately it is supposed to be tweaking the anything that is pool circuit dependent.

Before I make any more recommendations on this, how many relays in your panel are actually connected to something electrically?  Do I assume that there is a connection as follows.
Pool = IntelliChlor wired up to relay
North Jets = Connected to a single transformer were only the north side laminars are connected.
South Jets = Connected to a single transformer where only the south side laminars are connected
North Lights = Connected only to the north pool lights (A transformer if they are 12v lights)
South Lights = Connected only to the south pool lights (A transformer if they are 12v lights)
Waterslide = Connected to an aux booster pump for the slide.

Is the above the case?

**Reply by lockeandev** [reply]
Sorry it has taken so much time for me to respond. I have had a bunch of other issues with other projects and have not had time to debug this system. The good news is that my schedules appear to be running correctly. I have not noticed them failing to start over the past 4 weeks.  

However I am still happy to have suggestions for how to better wire this system up and run it. You assumptions about how my system currently operates are all correct except for the Waterslide. it is also run from the filter pump, no booster pump. And I can't remember how the intellichlor is hooked up exactly. It is connected to the built in salt cell wiring port on the bottom of the intellicenter so I believe it is powered from the intellicenter 12v transformer. I don't think that is connected to the Aux relays. There must be a low voltage relay on the intellicenter control board.

**Reply by rstrouse** [reply]
The top left relay in the bottom of your IntelliCenter load center is the filter relay.  The transformer for the IntelliChlor should be powered off the load side this relay.  The reason for this is because you do not want IntelliChlor to be able to energize the cell if the filter relay is not engaged.

I assume since you do not have a separate booster pump for the waterslide there is an automated valve actuator that redirects the flow to the waterslide.  Is that correct?  Also I see a waterfall pump listed.  What does that pump control?

**Reply by lockeandev** [reply]
Not sure what the waterfall is.  I only have the three water features North Jets, South Jets, and Waterslide. I have 4 intellivalves as follows:

1) Returns
2) North Jets
3) South Jets
4) Waterslide

**Comment by lockeandev** [comment]
The Intellichlor is connected to the socket at the bottom left of the intellicenter. the transformer in my cabinet is on the bottom right (I have the intellicenter with built in subpanel).
![image](https://user-images.githubusercontent.com/11232490/187263141-2e864149-ec30-47ec-b2e1-7c4e10cf16fb.png)

**Reply by rstrouse** [reply]
See page 36 of this manual.  Your IntelliChlor power supply should be wired as indicated.  This ensures IntelliChlor has no power unless the filter relay is engaged.
https://www.pentair.com/content/dam/extranet/nam/pentair-pool/residential/automation/intellicenter/manuals---user-docs/intellicenter-installation-guide.pdf


I mis-typed (misspoke) about the waterfall pump.   I saw the screenshot @tagyoureit and assumed that you had 2 pumps.   The unfortunate part about having only one pump is that it becomes very difficult to divert the proper amount of flow to each of the water features.  Are the north and south jets laminars or are they simply deck jets?   The slide requires a whole lot more flow than the jets so if you tune the pump for the jets and the slide together would probably the best speed setting for the slide.  More flow there is less noticeable.

However, since you have two sets of jets this gets a bit more complicated.  Having them plumbed separately makes coming up with a pump speed for the jets very difficult since small rpm changes likely change the performance of the jet a lot.  I would almost set up a circuit group and trigger the jets so that they all come on and set a speed for that.

**Comment by lockeandev** [comment]
I'm 95% sure the intellichlor is wired that way. I can open it up and check, but I know the intellichlor only comes on when the "pool" circuit is on. 

The jets are all laminars. 

I have the flowrates below figured out for my various combinations  of features. 

Pool only = 30 gpm (minimum flow for the salt cell and heater to run)
slide only  = 20 gpm (any more and it flows over the side of the slide) 
south jets = 20 gpm
north jets = 20 gpm
north or south jets and slide = 30 gpm
both north and south jets and slide = 40 gpm

I have circuit groups for each combination of features and the group includes a pump circuit for the applicable flow.

**Reply by rstrouse** [reply]
On your slide valve you should reduce the diversion of the valve so that your 30gpm output allows the slide to be operated without overflow.   This way the pool circuit will be the lowest.   The same thing should be set up for the north and south jets.  This will save you from blowing out the screens in the laminars.

The desire would be that the lowest flow would be for the pool only.  Then as you add features these would get additional flow.  I would set this up by controlling the diversion through the valves.  I suspect that right now you divert most or all of the flow to the specified feature.  One day when we get the promised control of IntelliValve (if Pentair ever gets this together) the control will be much more refined.

**Comment by lockeandev** [comment]
Yes that is the fan I have. I did not change any jumpers. I tried adding
address 3 and setting it as a fan, but then I don't get any info from the
fan reported to remchem.
